docs/upgrade-notes.asciidoc

```diff
@@ -0,0 +1,111 @@
+ifdef::env-github[]
+****
+link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-upgrade-notes.html[View this document on the Elastic website]
+****
+endif::[]
+[id="{p}-upgrade-notes"]
+== Upgrade notes
```

Comment:
Would calling this "Upgrading ECK" be more explicit in the right menu? ("upgrade notes" may sound like additional information for upgrading Elasticsearch)